sources/org.osbuild.containers-storage

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python3
+"""Provide access to an image in the host system's container storage.
+
+This stage differs from other sources in that the source check's to see if
+the container is available in the host's container storage. This puts a requirement
+on the user to ensure that the container is copied into local storage before trying
+to build an image. The starts by reading the host's `/etc/containers/storage.conf`
+file and then using the config to check if the container has been imported.
+
+Buildhost commands used: `skopeo`.
+"""
+
+import hashlib
+import subprocess as sp
+import sys
+
+from osbuild import sources
+from osbuild.util import containers
+
+SCHEMA = """
+"additionalProperties": false,
+"definitions": {
+  "item": {
+    "description": "The container image to fetch indexed by the container image id",
+    "type": "object",
+    "additionalProperties": false,
+    "patternProperties": {
+      "sha256:[0-9a-f]{64}": {
+        "type": "object",
+        "additionalProperties": false,
+        "required": ["image"],
+        "properties": {
+          "image": {
+            "type": "object",
+            "additionalProperties": false,
+            "required": ["name", "digest"],
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "Name of the image (including registry)."
+              },
+              "digest": {
+                "type": "string",
+                "description": "Digest of image in registry.",
+                "pattern": "sha256:[0-9a-f]{64}"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+},
+"properties": {
+  "items": {"$ref": "#/definitions/item"},
+  "digests": {"$ref": "#/definitions/item"}
+},
+"oneOf": [{
+  "required": ["items"]
+}, {
+  "required": ["digests"]
+}]
+"""
+
+
+class ContainersStorageSource(sources.SourceService):
```

Comment:
Fwiw,I am trying to write a small test for this source module in https://github.com/osbuild/osbuild/compare/main...mvo5:ocd-containers-storage-tests?expand=1  - I am currently struggling with the fact that I do not have a `/etc/containers/storage.conf` on my systems, hints appreciated :)

Reply:
Ah right, let me look into this. You're right, we should handle it. I think there are some fallback locations that we should maybe handle too

Reply:
Yeah the default, according to [the man pages](https://github.com/containers/storage/blob/main/docs/containers-storage.conf.5.md#files), is `/use/share/containers/storage.conf`

Reply:
Achilleas and I were just brainstorming and maybe a slightly easier way is to just call `podman info --format=json` because that will take care of reading from the multiple locations and the merging of that etc.

Reply:
Yeah and we won't need the TOML dependency. Amazing suggestion

Reply:
And then we discovered that we currently not depend on "podman" so we should probably not do it (just yet).